runtime/lua/vim/diagnostic.lua

```diff
@@ -1558,10 +1576,21 @@ M.handlers.virtual_text = {
 
     local virt_text_ns = ns.user_data.virt_text_ns
     local buffer_line_diagnostics = diagnostic_lines(diagnostics)
+
     for line, line_diagnostics in pairs(buffer_line_diagnostics) do
       if severity then
         line_diagnostics = filter_by_severity(severity, line_diagnostics)
       end
+
+      if opts.severity_sort then
+        table.sort(line_diagnostics, function(a, b)
+          if type(opts.severity_sort) == 'table' and opts.severity_sort.reverse then
+            return a.severity < b.severity
+          end
+          return a.severity > b.severity
+        end)
+      end
+
```

Comment:
These are already sorted in `vim.diagnostic.show` https://github.com/neovim/neovim/blob/c8e47f648052f8001d8a493cf7ca6c4867a36bc3/runtime/lua/vim/diagnostic.lua#L1755-L1765

Reply:
You're right, virtual text seems to be sorted already even without these changes, my bad.

Not sure how I didn't see this :sweat_smile: